AirCastingTests/Logic/SDSync/SDSyncFileWritingServiceTests.swift

```diff
@@ -0,0 +1,81 @@
+// Created by Lunar on 19/11/2021.
+//
+
+import XCTest
+@testable import AirCasting
+
+fileprivate extension SDCardSessionType {
+    var name: String {
+        switch self {
+        case .mobile: return "MOBILE"
+        case .fixed: return "FIXED"
+        case .cellular: return "CELLULAR"
+        }
+    }
+}
+
+class SDSyncFileWritingServiceTests: ACTestCase {
+    let service = SDSyncFileWritingService(bufferThreshold: 3)
+    let measurementsCountSession1 = 100
+    let measurementsCountSession2 = 80
+    let uuid = "123"
+    let uuid2 = "456"
+    
+    override func setUp() {
+        super.setUp()
+        SDCardSessionType.allCases.forEach { type in
+            for i in 0...measurementsCountSession1 {
+                let line = createFileLine(uuid: uuid + type.name, lineNumber: i, sessionType: type)
+                service.writeToFile(data: line, sessionType: type)
+            }
+            
+            for i in 0...measurementsCountSession2 {
+                let line = createFileLine(uuid: uuid2 + type.name, lineNumber: i, sessionType: type)
+                service.writeToFile(data: line, sessionType: type)
+            }
+        }
+    }
+    
+    func test_createsMobileAndFixedDirectories() {
+        let directories = service.finishAndSave()
+        
+        let documents = FileManager.default.urls(for: .documentDirectory, in: .allDomainsMask)[0]
+        let mobileFileURL = documents.appendingPathComponent("mobile")
+        let fixedFileURL = documents.appendingPathComponent("fixed")
+        XCTAssertTrue(FileManager.default.fileExists(atPath: mobileFileURL.path))
+        XCTAssertTrue(FileManager.default.fileExists(atPath: fixedFileURL.path))
+        XCTAssertEqual(directories.first?.0, mobileFileURL)
+        XCTAssertEqual(directories.last?.0, fixedFileURL)
+        XCTAssertEqual(directories.first?.1, .mobile)
+        XCTAssertEqual(directories.last?.1, .fixed)
```

Comment:
What would you say to mark this test as `throws` and then have some broder message when first will be nil?
`let firstDirectoryItem = try XCTUnwrap(directories.first, "Directories went wrong...")`